foreman/data_refinery_foreman/foreman/main.py

```diff
@@ -391,6 +407,182 @@ def retry_lost_processor_jobs() -> None:
         handle_processor_jobs(lost_jobs)
 
 
+##
+# Surveyors
+##
+
+@retry(stop_max_attempt_number=3)
+@transaction.atomic
+def requeue_survey_job(last_job: SurveyJob, dispatch=True) -> None:
+    """Queues a new survey job.
+
+    The new survey job will have num_retries one greater than
+    last_job.num_retries.
+    """
+    num_retries = last_job.num_retries + 1
+
+    new_job = SurveyJob(num_retries=num_retries,
+                        source_type=last_job.source_type
+                    )
+    new_job.save()
+
+    keyvalues = SurveyJobKeyValue.objects.filter(survey_job=last_job)
+
+    for keyvalue in keyvalues:
+        SurveyJobKeyValue.objects.get_or_create(survey_job=new_job,
+                                                key=keyvalue.key,
+                                                value=keyvalue.value,
+                                            )
+
+    try:
+        logger.info("Requeuing SurveyJob which had ID %d with a new SurveyJob with ID %d.",
+                    last_job.id,
+                    new_job.id)
+        try:
+            nomad_host = get_env_variable("NOMAD_HOST")
+            nomad_port = get_env_variable("NOMAD_PORT", "4646")
+            nomad_client = Nomad(nomad_host, port=int(nomad_port), timeout=5)
+            if dispatch:
+                nomad_response = nomad_client.job.dispatch_job("SURVEYOR", meta={"JOB_NAME": "SURVEYOR",
+                                                                                "JOB_ID": str(job.id)})
```

Comment:
Maybe we should switch surveyor jobs over to be expecting these meta values, however they are not currently what is accepted by surveyor jobs. They just need an accession.